### ISSUE TITLE
Unable to install PHP 5.6 /7.0 on Ubuntu 12.04 LTS

### ISSUE THREAD

**Issue by sulemanrai** [oerdnj/deb.sury.org#313]
I am trying to install php v5.6 or 7.0 . However i am not being successful. I am following the below steps on ubuntu 12.04 LTS 

```
sudo apt-get install software-properties-common
sudo add-apt-repository ppa:ondrej/php5-5.6
sudo apt-get update
sudo apt-get upgrade
sudo apt-get install php5
```

I am getting the following message 

apt-get install php5
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

```
The following packages have unmet dependencies:
 php5 : Depends: libapache2-mod-php5 (>= 5.6.19+dfsg-1+deb.sury.org~precise+1~) but 5.3.10-1ubuntu3.21 is to be installed or
                 libapache2-mod-php5filter (>= 5.6.19+dfsg-1+deb.sury.org~precise+1~) but it is not going to be installed or
                 php5-cgi (>= 5.6.19+dfsg-1+deb.sury.org~precise+1~) but it is not going to be installed or
                 php5-fpm (>= 5.6.19+dfsg-1+deb.sury.org~precise+1~) but it is not going to be installed
        Depends: php5-common (>= 5.6.19+dfsg-1+deb.sury.org~precise+1~) but 5.3.10-1ubuntu3.21 is to be installed
E: Unable to correct problems, you have held broken packages.
```

**Comment by quentin-berlemont** [comment]
I use this to get php7 cli :

```
$ sudo apt-get install software-properties-common
$ sudo add-apt-repository ppa:ondrej/php
$ sudo apt-get update
$ sudo apt-get install php php-cli
```

`$ php -v`

7.0.6-dev

**Comment by sulemanrai** [comment]
I am getting 

```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package php
E: Unable to locate package php-cli
```

**Comment by quentin-berlemont** [comment]
Did you add this one **ppa:ondrej/php** ?

**Comment by oerdnj** [comment]
PHP 7.0 is not supported on Ubuntu 12.04 and won't be. Also Ubuntu Precise support will be deprecated with Ubuntu 16.04 release, so it's time to update your Ubuntu installation. 

Paste output of:

`apt-cache policy php5`

and

`apt-cache policy php5-cli`

here.

**Comment by baskan** [comment]
I am getting this error on ubuntu 14.04 LTS
`
Creating config file /etc/php/7.0/mods-available/readline.ini with new version
Setting up php7.0-cli (7.0.5-1+deb.sury.org~trusty+1) ...
update-alternatives: using /usr/bin/php7.0 to provide /usr/bin/php (php) in auto mode
update-alternatives: using /usr/bin/phar7.0 to provide /usr/bin/phar (phar) in auto mode
update-alternatives: using /usr/bin/phar.phar7.0 to provide /usr/bin/phar.phar (phar.phar) in auto mode
Error: The new file /usr/lib/php/7.0/php.ini-production.cli does not exist!
dpkg: error processing package php7.0-cli (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of php7.0-fpm:
 php7.0-fpm depends on php7.0-cli; however:
  Package php7.0-cli is not configured yet.

dpkg: error processing package php7.0-fpm (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of php7.0:
 php7.0 depends on php7.0-fpm | libapache2-mod-php7.0 | php7.0-cgi; however:
  Package php7.0-fpm is not configured yet.
  Package libapache2-mod-php7.0 is not installed.
  Package php7.0-cgi is not installed.

dpkg: error processing package php7.0 (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of php:
 php depends on php7.0; however:
  Package php7.0 is not configured yet.

dpkg: error processing package php (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of php-cli:
 php-cli depends on php7.0-cli; however:
  Package php7.0-cli is not configured yet.

dpkg: erroNo apport report written because the error message indicates its a followup error from a previous failure.
                                                                                                                    No apport report written because the error message indicates its a followup error from a previous failure.
                                                       No apport report written because MaxReports is reached already
                                                                                                                     No apport report written because MaxReports is reached already
            r processing package php-cli (--configure):
 dependency problems - leaving unconfigured
Processing triggers for libc-bin (2.19-0ubuntu6.7) ...
Errors were encountered while processing:
 php7.0-cli
 php7.0-fpm
 php7.0
 php
 php-cli
E: Sub-process /usr/bin/dpkg returned an error code (1)
`

**Comment by baskan** [comment]
and these are the policy outputs:
```
root@:~# apt-cache policy php5
php5:
  Installed: (none)
  Candidate: 5.5.9+dfsg-1ubuntu4.14
  Version table:
     5.5.9+dfsg-1ubuntu4.14 0
        500 http://tr.archive.ubuntu.com/ubuntu/ trusty-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu/ trusty-security/main amd64 Packages
     5.5.9+dfsg-1ubuntu4 0
        500 http://tr.archive.ubuntu.com/ubuntu/ trusty/main amd64 Packages
root@:~#
root@:~# apt-cache policy php5-cli
php5-cli:
  Installed: (none)
  Candidate: 5.5.9+dfsg-1ubuntu4.14
  Version table:
     5.5.9+dfsg-1ubuntu4.14 0
        500 http://tr.archive.ubuntu.com/ubuntu/ trusty-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu/ trusty-security/main amd64 Packages
     5.5.9+dfsg-1ubuntu4 0
        500 http://tr.archive.ubuntu.com/ubuntu/ trusty/main amd64 Packages
root@:~#

```

**Comment by oerdnj** [comment]
@baskan don't paste the errors into random other bugs. I am going to delete those, as it's being already trackerd in #315

**Comment by oerdnj** [comment]
No response for 10 days, closing.

**Comment by harrisoftware** [comment]
I have this same problem

here is my output of apt-cache policy

php5:
  Installed: 5.5.26+dfsg-1+deb.sury.org~precise+1
  Candidate: 5.6.99.20160616+deb.sury.org~precise+1
  Version table:
     5.6.99.20160616+deb.sury.org~precise+1 0
        500 http://ppa.launchpad.net/ondrej/php5-compat/ubuntu/ precise/main i386 Packages
 *** 5.5.26+dfsg-1+deb.sury.org~precise+1 0
        100 /var/lib/dpkg/status
     5.3.10-1ubuntu3.26 0
        500 http://ca.archive.ubuntu.com/ubuntu/ precise-updates/main i386 Packages
        500 http://security.ubuntu.com/ubuntu/ precise-security/main i386 Packages
     5.3.10-1ubuntu3 0
        500 http://ca.archive.ubuntu.com/ubuntu/ precise/main i386 Packages

**Comment by oerdnj** [comment]
Ubuntu 12.04 LTS has reached its end-of-life last month. Just move on...

**Comment by ravindrabhat6** [comment]
Hi,

I am trying upgrade from 5.3 to 5.6 and getting the same error .
> sudo apt-get -y install php5.6 php5.6-mcrypt php5.6-mbstring php5.6-curl php5.6-cli php5.6-mysql php5.6-gd php5.6-intl php5.6-xsl php5.6-zip
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package php5.6
E: Couldn't find any package by regex 'php5.6'
E: Unable to locate package php5.6-mcrypt
E: Couldn't find any package by regex 'php5.6-mcrypt'
E: Unable to locate package php5.6-mbstring
E: Couldn't find any package by regex 'php5.6-mbstring'
E: Unable to locate package php5.6-curl
E: Couldn't find any package by regex 'php5.6-curl'
E: Unable to locate package php5.6-cli
E: Couldn't find any package by regex 'php5.6-cli'
E: Unable to locate package php5.6-mysql
E: Couldn't find any package by regex 'php5.6-mysql'
E: Unable to locate package php5.6-gd
E: Couldn't find any package by regex 'php5.6-gd'
E: Unable to locate package php5.6-intl
E: Couldn't find any package by regex 'php5.6-intl'
E: Unable to locate package php5.6-xsl
E: Couldn't find any package by regex 'php5.6-xsl'
E: Unable to locate package php5.6-zip
E: Couldn't find any package by regex 'php5.6-zip'

trying to execute this command in ubuntu 14 and stuck with this error. please help.

Thanks,
Ravi

**Comment by oerdnj** [comment]
No supported Ubuntu release has PHP 5.3 to begin with.

**Comment by ravindrabhat6** [comment]
its an old instance and I am in the process of upgrading it and I have got that error.

**Comment by ravindrabhat6** [comment]
as I mentioned in the previous comment, its an Ubuntu 14.04 . but since the error message was same, I thought I will ask for help here.

**Comment by oerdnj** [comment]
The most probable cause is that you are not using the PPA at all.

**Comment by nithinkolekar** [comment]
If I update the Ubuntu from 12.04 LTS to 14.04/16.06 should it also update php?

Currently I have php5-fpm and apache 2.2.22 which was shipped with Linode with ubuntu 12.04 long back and I never find it necessary to update Ubuntu until I came to know php7 is minimum recommended now a days.

**Comment by ravindrabhat6** [comment]
As far as I remember , when u do upgrade from 12.04 to 14.04  it will be upgraded to PHP 5.5.9 . and when you  upgrade from ubuntu14 to 16, by default it will be PHP 7
